src/modbus.c

```diff
@@ -657,9 +658,32 @@ static int response_io_status(int address, int nb,
 }
 
 /* Build the exception response */
+#ifdef __GNUC__
 static int response_exception(modbus_t *ctx, sft_t *sft,
-                              int exception_code, uint8_t *rsp)
+                              int exception_code, uint8_t *rsp,
+                              unsigned perform_flush, const char* format, ...)
+                              __attribute__ ((format (printf, 6, 7)));
```

Comment:
I expanded this function because there was the pattern of

- print error (when in debug mode)
- sometimes call _sleep_response_timeout(ctx) and modbus_flush(ctx)
- call response_exception

In order to unify that behavior I put the functionality, which was same all over the different places into that function.